### ISSUE TITLE
Index out of boundary for hess_rho_kl_crust_mantle

### ISSUE THREAD

**Issue by ziyixi** [geodynamics/specfem3d_globe#737]
I am using the development version to do the kernel simulation. And sometimes there is a "floating invalid" error. Than I enable the debug mode, and find the following error:
```
forrtl: severe (408): fort: (2): Subscript #3 of the array HESS_RHO_KL_CRUST_MANTLE has value 6 which is greater than the upper bound of 5

Image              PC                Routine            Line        Source
xspecfem3D         00000000010432D0  Unknown               Unknown  Unknown
xspecfem3D         00000000007B2B54  compute_kernels_h         954  compute_kernels.F90
xspecfem3D         00000000007B5AEF  compute_kernels_           63  compute_kernels.F90
xspecfem3D         00000000008D7DF6  iterate_time_undo         525  iterate_time_undoatt.F90
xspecfem3D         000000000040CA17  MAIN__                    486  specfem3D.F90
xspecfem3D         000000000040C77E  Unknown               Unknown  Unknown
libc-2.17.so       00002B46D66A63D5  __libc_start_main     Unknown  Unknown
xspecfem3D         000000000040C669  Unknown               Unknown  Unknown
```
This problem is related to https://github.com/geodynamics/specfem3d_globe/blob/4584d4d71204f6ec60d63e07a8d14aa6cfff614f/src/specfem3D/compute_kernels.F90#L911-L960

I print the values of `i, j, k` in line 953, and find they are all 6.

I am not familiar with the algorithm here and don't know if the part after the `i, j, k` loop should be put inside the loop or not. If it is the bug, sometimes the kernel calculation will get crashed, and sometimes not, which is common to see in the "out of boundary" kind of bug.

**Comment by danielpeter** [comment]
thanks :) 

you're right, it doesn't make much sense to have the `hess_rho_kl_crust_mantle(..)` array filled outside the i/j/k-do loops. depending on compilers, that will have the i/j/k values increased to 6 - thus, a nice feature to have it checked. what compiler did you use? 

...will be fixed in the next PR.

**Comment by danielpeter** [comment]
fixed by PR #738

**Comment by ziyixi** [comment]
Thanks for fixing the bug! I am using intel/18.0.2 on Stampede2.
